bundlerEnv: fix wrapping of programs

zimbatm opened the pull request:
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Setting the GEM_PATH after ruby is started is not reliable enough. In
some cases rubygems would have already loaded and ignore these settings.

Fixes #14048

cstrahan commented:
You might want to use the approach here (which I intend to merge soon):

https://github.com/NixOS/nixpkgs/pull/13975

zimbatm committed:
bundlerEnv: fix wrapping of programs

Setting the GEM_PATH after ruby is started is not reliable enough. In
some cases rubygems would have already loaded and ignore these settings.

Fixes #14048

zimbatm commented:
Nice, fixed

cstrahan commented:
Cool - I think that looks good to me.
